src/main/java/io/fabric8/jenkins/openshiftsync/BuildSyncRunListener.java

```diff
@@ -201,9 +206,15 @@ private void upsertBuild(Run run, RunExt wfRunExt) {
       return;
     }
 
+    BluePipeline blueResource = (BluePipeline) BluePipelineFactory.resolve(run.getParent());
     String rootUrl = OpenShiftUtils.getJenkinsURL(getOpenShiftClient(), cause.getNamespace());
     String buildUrl = joinPaths(rootUrl, run.getUrl());
     String logsUrl = joinPaths(buildUrl, "/consoleText");
+    String logsConsoleUrl = joinPaths(buildUrl, "/console");
+    String logsBlueOceanUrl = null;
+    if (blueResource != null)
+        logsBlueOceanUrl = joinPaths(rootUrl, "/blue", "/organizations", "/" + blueResource.getOrganization(), "/" + run.getParent().getName(), "/detail",
+                "/" + run.getParent().getName(), "/" + run.getNumber(), "/pipeline");
```

Comment:
Can any of these values have special characters that need to be encoded in the URL?

It also looks like `joinPaths` will add the separators (if I'm looking at the right source), so they shouldn't be needed here.